core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -1111,6 +1111,10 @@ class ZKCheckedEphemeral(path: String,
                             case Code.INVALIDACL =>
                               error("Invalid ACL")
                               setResult(Code.INVALIDACL)
+                            case Code.NOAUTH =>
+                              // Allows old consumer to register with secure ZK
+                              error("NOAUTH raised on path %s (AUTH may not be needed to create children)".format(path))
+                              createRecursive(path, suffix)
```

Comment:
@hachikuji, you are correct, I misinterpreted that thinking it stepped to the next node. 

In retrospect I think we could just ignore this here and let the CreateCallback handle any lingering issues. 

I foresee the following issues with ignoring the NOAUTH here: 

1. Turns out we needed to create this node and its actually non-existent 
2. We don't actually have permission to create the ephemeral node(the leaf node)
3. It will be harder to troubleshoot since we keep walking the path 

Items 1 and 2 I believe will be caught by the CreateCallback and/or GetDataCallback. I think logging the error handles troubleshooting issues.